[DOCS] New page: Best practices for querying alerts (#1527)
elastic/security-docs: docs/detections/query-alert-indices.asciidoc

[[query-alert-indices]]
== Query alert indices
This page explains how you should query alert indices, for example when building rule queries, custom dashboards, or visualizations. For more information about alert event field definitions, review the [Alert schema].

[discrete]
=== Alert indices
Alert events are stored in hidden {es} indices. The naming conventions for these indices and their aliases differ depending on which version of {stack} you're running:

* *8.0.0 and later:* `.internal.alerts-security.alerts-<space-id>-NNNNNN`, where `NNNNNN` is a number that increases over time, starting from 000001.

> * *Before 8.0.0:* `.siem-signals-<space-id>-NNNNNN`, where `NNNNNN` is a number that increases over time, starting from 000001.

Reviewer: I would consider moving this example of how **not** to query the alerts indices down below the recommended method. If a user stops reading the page after this section, they have all the information needed to query the alert indices incorrectly and haven't reached any text warning them that these are actually not the recommended ways.

It might also be helpful to include a warning within this section that this section is provided only for context and to provide a concrete example of how correct and incorrect usages differ.

Author: Great feedback, thanks.

[discrete]
=== Alert index aliases
We recommend querying the following index aliases to avoid querying the numbered alert indices:

* If you're using version 8.0.0 or later: `.alerts-security.alerts-<space-id>`
This alias includes both the legacy (before 8.0.0) and the new alert indices.

> * If you're using a version before 8.0.0: `.siem-signals-<space-id>`

Reviewer 2: Consider either changing the first part to "8.0.0 and later:" and "Before 8.0.0:" to match the previous section, OR change the previous section to match the phrasing in this list. No strong preference either way, other than try to use parallel form for both lists.

> Queries on this alias will function regardless of your {stack} version, but will not follow the newer `.alerts` naming convention, and may be deprecated in future releases.

Reviewer 3: ```suggestion
Queries on this alias will function regardless of your {stack} version, but will not follow the newer `.alerts` naming convention and may be deprecated in future releases.
```

Reviewer 2: The first comma ("version, but") is also grammatically incorrect, but prob OK to keep if you feel a comma pause is warranted

[[query-alert-indices]]
[role="xpack"]
== Query alert indices

> This page explains how you should query alert indices, for example when building rule queries, custom dashboards, or visualizations. (For more information about alert event field definitions, review the [Alert schema].)

Reviewer 4: Agree ^.

> This page explains how you should query alert indices, for example when building rule queries, custom dashboards, or visualizations. (For more information about alert event field definitions, review the [Alert schema].)

Reviewer 4: ```suggestion
This topic explains how you should query alert indices, for example, when building rule queries, custom dashboards, or visualizations. (For more information about alert event field definitions, review the [Alert schema].)
```

Alert events are stored in hidden {es} indices. The naming conventions for these indices and their aliases differ depending on which version of {stack} you're running:

*Alerts from {stack} versions before 8.0*

> The naming convention is `.siem-signals-<space id>-NNNNNN` where `NNNNNN` is a number that counts up over time, starting from 000001.

Reviewer 4: I think we should put a comma after the formatting. 
```suggestion
The naming convention is `.siem-signals-<space id>-NNNNNN`, where `NNNNNN` is a number that counts up over time, starting from 000001.
```

> *Alerts from {stack} versions 8.0 and later*

Reviewer 3: ```suggestion
*Alerts from {stack} versions 8.0.0 and later*
```

> The naming convention is `.internal.alerts-security.alerts-<space id>-NNNNNN` where `NNNNNN` is a number that counts up over time, starting from 000001.

Reviewer 4: ```suggestion
The naming convention is `.internal.alerts-security.alerts-<space id>-NNNNNN`, where `NNNNNN` is a number that increases over time, starting from 000001.
```

You can query the following index aliases to avoid querying the numbered alert indices:

[discrete]
=== Best practices
If you're using a version before 8.0, you should query the following index alias:

> * `.siem-signals-<space id>` (Note that there is no dash or wildcard after the space ID).

Author: I like having it in both places so that the instructions are skim-proof.

For example, to query all spaces: `.siem-signals-*`.

> Once you update to 8.0 or later, you should query the following index alias, which will include both legacy (before 8.0) and new alerts:

Reviewer 3: ```suggestion
Once you update to 8.0.0 or later, you should query the following index alias, which will include both legacy (before 8.0.0) and new alerts:
```

Reviewer 4: ```suggestion
Once you upgrade to 8.0 or later, you should query the following index alias, which will include both legacy (before 8.0) and new alerts:
```

* `.alerts-security.alerts-<space id>` (Note that there is no dash or wildcard after the space ID).

> For example, to query all spaces: `.alerts-security.alerts-*`.

Reviewer 3: ```suggestion
For example, use `.alerts-security.alerts-*` to query all spaces.
```

[discrete]
=== Acceptable practices
Queries on the following index aliases will function regardless of your {stack} version, but will not follow the `.alerts` naming convention and may be deprecated in future releases:

* `.siem-signals-<space ID>` (Note that there is no dash or wildcard after the space ID).

> For example, to query all spaces: `.siem-signals-*`.

Reviewer 3: ```suggestion
For example, use `.siem-signals-*` to query all spaces.
```

[discrete]
== Alert aliases by {stack} version

> Here is a complete list of the available alert index aliases, by version.

Reviewer 3: ```suggestion
The following sections contain a complete list of the available alert index aliases, by version.
```

[discrete]

> === For alerts from versions before 8.0:

Reviewer 3: ```suggestion
=== For alerts from versions before 8.0.0:
```

* `.siem-signals-<space id>`
* `.alerts-security.alerts-<space id>` (Only available after upgrading to 8.0).
NOTE: Alerts from versions before 8.0 will stay saved in legacy alert indices, but can be queried using the new alias once you upgrade to 8.0.

[discrete]

> === For alerts from versions 8.0 and later:

Reviewer 3: ```suggestion
=== For alerts from versions 8.0.0 and later:
```
